### PR TITLE
Bodhi2 support!

### PR DESCRIPTION
**Pull request by lmacken**
This will automatically detect which version of bodhi it's querying and will
return the appropriate client.

### CODE REVIEW

File: fedora/client/bodhi.py [hunk 1]
```diff
@@ -24,22 +24,376 @@
...
+    if staging:
+        log.info('Using bodhi2 STAGING environment')
+        base_url = STG_BASE_URL
+        fedora.client.openidproxyclient.FEDORA_OPENID_API = STG_OPENID_API
+
+    log.debug('Querying bodhi API version')
+    api_url = base_url + 'api_version'
+    response = requests.head(api_url)
+    if response.status_code == 200:
+        print('Bodhi 2 detected')
```
Comment (ralphbean) [inline fedora/client/bodhi.py]: Do we want to remove these print statements before we ship this?  (or demote them to log.debug statements?)

File: fedora/client/bodhi.py [hunk 2]
```diff
@@ -24,22 +24,376 @@
...
 class BodhiClientException(FedoraClientError):
     pass
 
 
-class BodhiClient(BaseClient):
+def BodhiClient(base_url=BASE_URL, staging=False, **kwargs):
+    if staging:
+        log.info('Using bodhi2 STAGING environment')
+        base_url = STG_BASE_URL
+        fedora.client.openidproxyclient.FEDORA_OPENID_API = STG_OPENID_API
```
Comment (ralphbean) [inline fedora/client/bodhi.py]: This ``if staging`` stanza appears here and also inside the ``Bodhi2Client`` init method.  It should probably appear only once (just inside the Bodhi2Client class, prolly).

File: fedora/client/bodhi.py [hunk 3]
```diff
@@ -24,22 +24,376 @@
...
+class Bodhi2Client(OpenIdBaseClient):
+
+    def __init__(self, base_url=BASE_URL, username=None, password=None,
+                 staging=False, **kwargs):
+        if staging:
+            log.info('Using bodhi2 STAGING environment')
+            base_url = STG_BASE_URL
+            fedora.client.openidproxyclient.FEDORA_OPENID_API = STG_OPENID_API
+        super(Bodhi2Client, self).__init__(base_url, login_url=base_url +
+                                          'login', debug=True, **kwargs)
```
Comment (ralphbean) [inline fedora/client/bodhi.py]: debug=True is hardcoded here.  Should we use **kwargs for it instead?

File: fedora/client/bodhi.py [hunk 4]
```diff
@@ -24,22 +24,376 @@
...
+
+    def latest_builds(self, package):
+        return self.send_request('latest_builds', params={'package': package})
+
+    def testable(self):
+        """ Get a list of installed testing updates.
+
+        This method is a generate that yields packages that you currently
+        have installed that you have yet to test and provide feedback for.
+
```
Comment (ralphbean) [inline fedora/client/bodhi.py]: Might want to add a sentence to this docstring indicating that this function won't work on a system without dnf.

File: fedora/client/bodhi.py [hunk 5]
```diff
@@ -24,22 +24,376 @@
...
+        builds = []
+        data = self.get_releases()
+        koji = self.get_koji_session(login=False)
+        for release in data['releases']:
+            try:
+                for build in koji.listTagged(release['candidate_tag'], latest=True):
+                    if build['owner_name'] == self.username:
+                        builds.append(build)
+            except:
+                pass
```
Comment (ralphbean) [inline fedora/client/bodhi.py]: We should log something here, prolly.  ``log.exception()`` maybe?

File: fedora/client/bodhi.py [hunk 6]
```diff
@@ -49,7 +403,7 @@ def __init__(self, base_url='https://admin.fedoraproject.org/updates/',
         :kwarg base_url: Base of every URL used to contact the server.
             Defaults to the Fedora Bodhi instance.
         :kwarg useragent: useragent string to use.  If not given, default to
-            "Fedora Boshi Client/VERSION"
+            "Fedora Bodhi Client/VERSION"
```
Comment (ralphbean) [inline fedora/client/bodhi.py]: clutch!

**Comment by ralphbean** [comment]
Couple comments, but it looks awesome.  Can merge soon!

File: fedora/client/bodhi.py (hunk 2 above)
Comment (lmacken) [inline fedora/client/bodhi.py]: Done. We need it in the `BodhiClient` method so we know which instance to query for the api version.

File: fedora/client/bodhi.py (hunk 4 above)
Comment (lmacken) [inline fedora/client/bodhi.py]: Done.

**Comment by lmacken** [comment]
All comments addressed. Thanks for the review!

**Comment by ralphbean** [comment]
:+1: after merge I can get a release going.

**Pull request merged**